### ISSUE TITLE
When use `binding.pry` once, Ruby process get slow.

### ISSUE THREAD

**Issue by kntmrkm** [deivid-rodriguez/pry-byebug#61]
Hi, 

This is a profile before use `binding.pry`.

![before](https://cloud.githubusercontent.com/assets/2703486/7648863/73d09784-fb22-11e4-94e6-6266d537441b.jpg)

After use `binding.pry` and quit it.
Page load time get slow. Webrick's ruby process got slow maybe.

![after](https://cloud.githubusercontent.com/assets/2703486/7648889/c5338a78-fb22-11e4-8bde-a3a45db09ab6.jpg)

Without `pry-byebug`.
Process does not get slow.

This is Gemfile then. and using ruby 2.2.0.

```ruby
source 'https://rubygems.org'

gem 'pry-rails'

group :development do
  gem 'rack-mini-profiler'
end

group :test do
  gem 'sqlite3'
end

group :development, :test do
  gem 'pry-byebug'
end

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem 'rails', '4.2.1'
gem 'pg'
gem 'sass-rails', '~> 5.0'
gem 'uglifier', '>= 1.3.0'
gem 'coffee-rails', '~> 4.1.0'
gem 'jquery-rails'
gem 'turbolinks'
gem 'jbuilder', '~> 2.0'
gem 'sdoc', '~> 0.4.0', group: :doc

group :development, :test do
  gem 'byebug'
  gem 'web-console', '~> 2.0'
  #gem 'spring'
end



```

**Comment by deivid-rodriguez** [comment]
Yep, debuggers makes things slower.

Do you find this too annoying? Not sure how much room there's for improvement and whether it is worth it.

**Comment by kntmrkm** [comment]
I don't know how much worth too.

At least It's a annoying for me.

In my main project.
Page load time is increase to about 15 seconds from few seconds.

Other people are how's it going?

**Comment by deivid-rodriguez** [comment]
@kntmrkm Can you confirm this is reproduced using byebug only, open an issue and byebug's repo and link to this one there? Thanks.

**Comment by NullVoxPopuli** [comment]
my page time goes from seconds to about a minute

**Comment by deivid-rodriguez** [comment]
Thanks guys for your reports.

I would just ask you to give objective numbers instead of biased stuff. "from x seconds to y seconds" is much more useful than "to about 15 seconds from few seconds" or "from seconds to about a minute".

**Comment by NullVoxPopuli** [comment]
Well, looking at my log, it looks like the response time after binding.pry is used ranges from 40 seconds to over a minute...

![screenshot from 2015-05-18 21 07 21](https://cloud.githubusercontent.com/assets/199018/7693593/ee822462-fda1-11e4-8ad1-80ea2b6db8a7.png)

**Comment by deivid-rodriguez** [comment]
Again not very specific, but got it. Thanks.

**Comment by NullVoxPopuli** [comment]
does it not happen for you?

**Comment by deivid-rodriguez** [comment]
Of course, debuggers slow things down, that's normal. But it never got too bad in my apps.

In any case, I have an idea on how to improve this.

**Comment by SimonHFrost** [comment]
This seems to increase our page load 10x. I did some analysis and have these load times from our home page:

<b>Before</b>
Completed 200 OK in 2317ms (Views: 2174.1ms)
Completed 200 OK in 2424ms (Views: 2283.9ms)
Completed 200 OK in 1624ms (Views: 1617.3ms)
Completed 200 OK in 1682ms (Views: 1667.9ms)
<b>After</b>
Completed 200 OK in 32467ms (Views: 32290.0ms)
Completed 200 OK in 18576ms (Views: 18498.0ms)
Completed 200 OK in 20572ms (Views: 20429.7ms)
Completed 200 OK in 19399ms (Views: 19326.8ms)

I've found that restarting the server is a lot faster than waiting for multiple page loads of this duration.

**Comment by jbielick** [comment]
Also experiencing this issue:

`> rails s`

```
Completed 200 OK in 1625ms (Views: 1062.8ms | ActiveRecord: 114.6ms)
Completed 200 OK in 319ms (Views: 232.7ms | ActiveRecord: 43.8ms)
Completed 200 OK in 313ms (Views: 228.3ms | ActiveRecord: 43.4ms)
Completed 200 OK in 330ms (Views: 241.1ms | ActiveRecord: 45.1ms)
Completed 200 OK in 301ms (Views: 216.4ms | ActiveRecord: 43.9ms)
```

**ruby ~2.0% CPU**

after inserting one `binding.pry`, hitting that breakpoint, removing it and refreshing the page:

```
Completed 200 OK in 7361ms (Views: 5164.5ms | ActiveRecord: 992.0ms)
Completed 200 OK in 7423ms (Views: 5236.2ms | ActiveRecord: 932.3ms)
Completed 200 OK in 7434ms (Views: 5160.5ms | ActiveRecord: 1062.8ms)
Completed 200 OK in 7481ms (Views: 5263.7ms | ActiveRecord: 876.3ms)
Completed 200 OK in 7296ms (Views: 5077.4ms | ActiveRecord: 986.1ms)
```
and
**ruby 99.7-100.4% CPU**


Relevant Gemfile group:

```ruby
group :development, :test do
  gem 'capybara', '~> 2.4.1'
  gem 'rspec-rails'
  gem 'rspec-activemodel-mocks'
  gem 'rspec-collection_matchers'
  gem 'jasmine'
  gem 'jasmine-headless-webkit'
  gem 'premailer'
  gem 'spring'
  # spring/rspec binstubs
  gem 'spring-commands-rspec'
  # rspec guard watcher for TDD
  gem 'guard-rspec', require: false
  gem 'pry', '~> 0.10.0'
  gem 'pry-rails'

  group :darwin do
    # filesystem api events in mac osx
    gem 'rb-fsevent'
  end
  group :pry do
    gem 'pry-byebug'
  end
end
```

lock:

```ruby
    pry (0.10.1)
      coderay (~> 1.1.0)
      method_source (~> 0.8.1)
      slop (~> 3.4)
    pry-byebug (3.1.0)
      byebug (~> 4.0)
      pry (~> 0.10)
    pry-rails (0.3.4)
      pry (>= 0.9.10)
```

**Comment by deivid-rodriguez** [comment]
Thanks, I'll fix this as soon as possible.

**Comment by fredrikmollerstrand** [comment]
Our team is seeing this as well: any requests to Rails after entering and exiting a `binding.pry` session is very slow with the process seemingly locking up for 5-10 seconds.

Let me know if I can be of any help chasing this down, @deivid-rodriguez.

**Comment by naw** [comment]
I'm seeing this too.

Before: 900ms
After: 13,500ms

**Comment by deivid-rodriguez** [comment]
Closing in favor of byebug#144
